src/connector/src/parser/plain_parser.rs

```diff
@@ -58,12 +67,23 @@ impl PlainParser {
 
     pub async fn parse_inner(
         &mut self,
-        payload: Vec<u8>,
+        key: Option<Vec<u8>>,
+        payload: Option<Vec<u8>>,
```

Comment:
why changing to `Option`

Reply:
cause if we want to parse key, we need to call `parse_one` and then call `parse inner`. The argument pass by `parse_one` is Option. Previously we don't need to parse key so we just get item inside the `Option<Payload>` in the function `parse_one` by macro. Currently we need to get item both `Option<Payload>` and `Option<Key>`. So I contend that getting item inside the option can be moved into function `parse_inner`